coupa/chosen
# JZ-27019 master Expose setValue interface of Chosen widget

## JIRA

* [Main JIRA ticket] https://coupadev.atlassian.net/browse/JZ-27019
* [JIRA propagation subtask] https://coupadev.atlassian.net/browse/JZ-27391

## Code reviewers
- [ ] @johnny-lai 

## Summary of issue
**The issue only happens when the lookup has more than 150 lookup values.**
a. Create lookup custom field and set it default from commodity(Let's say commodity_A).
b. In cart page, click `Add Line`. Search and select the item.
c. Selected item will default commodity to "commodity_A". After that the expected behavior is "commodity_A" should trigger another defaulting to the lookup custom field.

## Summary of change
Try to expose value set interface for `Chosen` widget.
After that we set the value by:
`$select_field.chosen(['setValue', value]);`

## Testing approach
1. Checked in to the clone instance : https://clone-adb-test-jz-26155.coupadev.com
2. Clicked on cart to check.
3. Click on add line > Choose item “STIRRER, WOODEN (DS21579)”
4. All the default values gets populated except the custom field “Asset Category”.
5. Save the line and hen click on billing “magnifying glass” to choose a account > click on cost centre “commercial service unit” > choose.
6. You will be able to see the asset category “EXPENSED.EQUIPT.FOODSERV.OTHERS” populated.


## Select which Feature Test set to run

_Please select which feature test set to run for this PR. If no option is selected, by default Speculator tests and Component based tests will run. Please note that "No tests" and "All tests" are mutually exclusive and only Speculator and Component tests can be combined together._

 - [x] Speculator tests
 - [x] Component tests
 - [ ] All tests

## Useful Links

* [Code Review Checklists 1](https://q.coupadev.com/topics/code-review-checklists)
* [Code Review Checklists 2](https://coupadev.atlassian.net/wiki/display/CD/Code+Review+Checklist)
* [Guidelines for Risk Levels](https://coupadev.atlassian.net/wiki/display/CD/Guidelines+for+Risk+Levels)
* [Second Level Review Rubric](https://q.coupadev.com/topics/second-level-review-rubric)
* [Common CR issues](https://q.coupadev.com/topics/second-level-review-rubric)
* [GitHub PR Comments 1](https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments)
* [GitHub PR Comments 2](https://github.com/blog/821)
* [The UI Integration Tests Code Review Checklist](https://coupadev.atlassian.net/wiki/spaces/CQ/pages/181212079/The+UI+Integration+Tests+Code+Review+Checklist)
